frontend/modules/compare/compare.sagas.ts

```diff
@@ -303,6 +294,7 @@ function* startComparisonOfFederation() {
 		}
 	}
 
+	yield setModelsNodesVisibility(modelsToReuse, VISIBILITY_STATES.VISIBLE);
```

Comment:
not quite what I meant... I think we need 2 arrays here. `modelsToShow`, `modelsToHide`

modelsToShow initialises with baseModels, ditch the calls on L267 L268
while we loop, we add the individual target models to `modelsToShow` or `modelsToHide` depending on whether we can reuse them.

at this point, we call setModelsNodesVisibility on the 2 arrays. Then we reduce this down to 2 calls to set visibility without switching off and on the ones we are to reuse.

I think this makes sense (in my head at least), but maybe I'm missing something?

Reply:
Done ;)